Update mega-drive-genesis.md: Grammar fixes and Clarifications (#124)
flipacholas/Architecture-of-consoles: articles/mega-drive-genesis.md

Game cartridges stores tiles in their **ROM** (found in their cartridge) but they have to be copied to **VRAM** so the VDP can read them {{< cite "graphics-ports" >}}. Traditionally, this was only possible during specific time frames and handled by the CPU, fortunately, this console added special circuitry to offload this task to the VDP (we'll get into details later on).

Tiles are used to build a total of **four planes** which, once merged together, form the frame seen on the screen. Also, planes' tiles will overlap with each other, so the VDP will decide which tile is going to be visible based on the type of plane and the tile's **priority value**.
{{% /inner_markdown %}}

{{< /tab >}}

{{< tab name="Background" >}}

{{< tabs nested="true" class="pixel" float="true" >}}
  {{< tab_figure_img name="Full" active="true" src="vdp_sonic/layer2.png" >}}
Allocated Background map.
  {{< /tab_figure_img >}}

>   {{< tab_figure_img name="Selected" src="vdp_sonic/layer2_selected.png" >}}

Reviewer: I think the original name is fine, the 2nd image is not cropped, just marked

Author: It's not cropped, yes, but I felt that using the same terminology used in the rest of the tabs (which are also not cropped) made more sense and created uniformity.

Reviewer: iirc sega uses the term 'plane' instead of 'layer', so this change would also need an overhaul on other articles as well.

Author: I don't think you fully understand: the article already uses 'Full' and 'Selected' in the tabs for the other planes (Foreground and Sprites) and all this is doing is making this tab (Background) match the same terminology used in the other two tabs about the other planes.

![image](https://user-images.githubusercontent.com/7586972/177685879-5c29d814-d087-44f3-b06e-f7db298f2fd7.png)
![image](https://user-images.githubusercontent.com/7586972/177685914-bbe23e55-c698-4b3f-986f-0d2fd624353c.png)

Author: If you're suggesting changing all the tabs to "Plane" and "Marked" that can also be done but I was simply going with the form I saw more ("Full" and "Selected").

Each tile can be **flipped horizontally and/or vertically** and have a **priority** set.

> On the showed example you'll notice that the selected area for display is not a square... *It doesn't have to be!*. The VDP allows to set up horizontal scrolling values for the whole frame, each individual scan-line or every eight pixels. This means that developers can shape the selected area like a rhomboid and alter its angles as the player moves to simulate **perspective effects**. Tricks like this one don't corrupt the plane, the VDP fetches each (selected) horizontal line and builds a regular frame from it.

Reviewer: is the 'be' grammatically required?

Author: I think it does; Without it I can definitely still tell what you're saying, but it feels a bit "odd" would be the best way to describe it. Perhaps it's not necessarily a matter of grammatical correctness, but rather just creating a better reading flow?

### Video Output

> The first major design of this console (commonly refered to as the Model 1) has the same video out port of [the Master System]({{< ref "master-system#video-output" >}}). The Model 2 and Model 3 use a mini-DIN port instead.

Reviewer: ```suggestion
The first design of this console (commonly referred to as the 'Model 1') has the same video out port of [the Master System]({{< ref "master-system#video-output" >}}). The subsequent 'Model 2' and 'Model 3' switched to a mini-DIN port instead.
```

Author: LGTM, can't believe I didn't catch my misspelling of referred.

Frequency modulation or 'FM' synthesis is one of many professional techniques used for synthesising sound, it significantly rose in popularity during the 80s and made way to completely new sounds (many of which you can find by listening to the pop hits from that era).

> In an *incredibly simplified* nutshell, the FM algorithm takes a single waveform (**carrier**) and alters its frequency using another waveform (**modulator**), and the result is a new waveform with a different sound. The carrier-modulator combination is called **operator** and multiple operators can be chained together to form the final waveform. Different combinations achieve different results. This chip allows 4 operators per channel.

Reviewer: I would add a period instead of an 'and' (just realised there's too many ands)

Author: I'll commit that instead, agreed that it flows better in this case.

> In an *incredibly simplified* nutshell, the FM algorithm takes a single waveform (**carrier**) and alters its frequency using another waveform (**modulator**), and the result is a new waveform with a different sound. The carrier-modulator combination is called **operator** and multiple operators can be chained together to form the final waveform. Different combinations achieve different results. This chip allows 4 operators per channel.

Reviewer: ```suggestion
In an *incredibly simplified* nutshell, the FM algorithm takes a single waveform (**carrier**) and alters its frequency using another waveform (**modulator**). The result is a new waveform with a different sound. The carrier-modulator combination is called **operator** and multiple operators can be chained together to form the final waveform. Different combinations achieve different results. This chip allows 4 operators per channel.
```

This is actually the original Master System's [sound chip]({{< ref "master-system#audio" >}}) and it's embedded in the VDP. It runs at the speed of the Z80.

> Notice the 'Pulse 3' channel remains unused. This is because the game uses a mode for the noise channel that reserves the third pulse channel for modulation {{< cite "audio-sonic" >}}, a function which was featured on the Master System as well.

Reviewer: ```suggestion
Notice the 'Pulse 3' channel remains unused. This is because the game uses a mode for the noise channel that reserves the third pulse channel for modulation {{< cite "audio-sonic" >}}, a function featured on the Master System as well.
```
